lib/metric-config-parser/metric_config_parser/tests/test_experiment.py

```diff
@@ -267,6 +267,46 @@ def test_preenrollment(self, experiments, config_collection):
         assert len(days28_metrics) == 1
         assert days28_metrics[0].metric.name == "spam"
 
+    def test_segment_resolution(self, experiments, config_collection):
+        spec = AnalysisSpec()
+
+        cfg = spec.resolve(experiments[9], config_collection)
+        segments = cfg.experiment.segments
+
+        assert any(
+            segment.name == "regular_users_v3" and segment.data_source.name == "my_cool_data_source"
+            for segment in segments
+        )
+
+    def test_segment_definition_and_experiment_segment(self, experiments, config_collection):
+        config_str = dedent(
+            """
+            [experiment]
+            segments = ["test_segment"]
+
+            [segments.test_segment]
+            Data_Source = "test_data_source"
+            Select_Expression = "{{agg_any('1')}}"
+
+            [segments.data_sources.test_data_source]
+            from_expression = "(SELECT 1 WHERE submission_date BETWEEN {{experiment.start_date_str}} AND {{experiment.last_enrollment_date_str}})"
+            """  # noqa
+        )
+        spec = AnalysisSpec.from_dict(toml.loads(config_str))
+        cfg = spec.resolve(experiments[9], config_collection)
```

Comment:
Great test! This actually raises an important question -- should the custom experiment config override or augment the default segments from Experimenter? This isn't an issue for outcomes because we don't support outcomes in custom experiment configs -- they're only defined via Experimenter. However, the precedent here that we can draw from is metric analysis periods, and in that case we would merge them from various sources to get a final list containing metrics from all sources (defaults, outcomes, custom). So for segments, we can do the same -- we'll always use the segments from Experimenter, and if there are segments in a custom config we'll add those on as well.

Mostly documenting this for posterity. We should probably also add this to the docs, though after a quick look I'm not sure exactly the best place for it. We can take a closer look later and see where makes the most sense.